src/core/qfieldcloudprojectsmodel.h

```diff
@@ -496,6 +496,7 @@ class QFieldCloudProjectsModel : public QAbstractListModel
     void downloadFileConnections( const QString &projectId, const QString &fileName );
     void loadProjects( const QJsonArray &remoteProjects = QJsonArray(), bool skipLocalProjects = false );
     void insertProjects( const QList<CloudProject *> &projects );
+    void logFailedDownload( CloudProject *projectName, const QString &projectId, const QString &fileName, const QString &errorMessage, const QString &errorMessageDetail );
```

Comment:
```suggestion
    void logFailedDownload( CloudProject *project, const QString &fileName, const QString &errorMessage, const QString &errorMessageDetail );
```

Note you should rename `projectName` to `project`, as it has the `CloudProject` instance (and you use that name in the `.cpp` file). Also no need for explicit `projectId` param, as it is already in the `CloudProject` instance. Even if it is not (I did not check), please add it there.

Reply:
There is `project->id` i think that would work.